pkg/operator/status.go

```diff
@@ -161,9 +162,19 @@ func (optr *Operator) syncDegradedStatus(ierr error) (err error) {
 
 		// set progressing
 		if cov1helpers.IsStatusConditionTrue(co.Status.Conditions, configv1.OperatorProgressing) {
-			cov1helpers.SetStatusCondition(&co.Status.Conditions, configv1.ClusterOperatorStatusCondition{Type: configv1.OperatorProgressing, Status: configv1.ConditionTrue, Message: fmt.Sprintf("Unable to apply %s", optrVersion)})
+			cov1helpers.SetStatusCondition(&co.Status.Conditions, configv1.ClusterOperatorStatusCondition{
+				Type:    configv1.OperatorProgressing,
+				Status:  configv1.ConditionTrue,
+				Message: fmt.Sprintf("Unable to apply %s", optrVersion),
+				Reason:  reason,
```

Comment:
This does look good but I think we need to do a better job into at least saying which of the sync function failed

Reply:
I'll open a follow up

Reply:
addressing this here directly